src/sql_utils.h

```diff
@@ -0,0 +1,123 @@
+  bindArgument<const char*>(db, statement, cnt, v.c_str());
+}
+
+struct SQLBlob {
+  const std::string& content;
+  SQLBlob(const std::string& str) : content(str) {}
+};
+
+template <typename T>
+static void bindArgument(sqlite3* db, sqlite3_stmt* statement, int cnt, const SQLBlob& blob) {
+  if (sqlite3_bind_blob(statement, cnt, blob.content.c_str(), blob.content.size(), NULL) != SQLITE_OK) {
+    LOG_ERROR << "Could not bind: " << sqlite3_errmsg(db);
+    throw std::runtime_error("SQLite bind error");
+  }
+}
+
+struct SQLZeroBlob {
+  size_t size;
+};
+
+template <typename T>
+static void bindArgument(sqlite3* db, sqlite3_stmt* statement, int cnt, const SQLZeroBlob& blob) {
+  if (sqlite3_bind_zeroblob(statement, cnt, blob.size) != SQLITE_OK) {
+    LOG_ERROR << "Could not bind: " << sqlite3_errmsg(db);
+    throw std::runtime_error("SQLite bind error");
+  }
+}
+
+template <typename... Types>
+static void bindArguments(sqlite3* db, sqlite3_stmt* statement, int cnt) {
+  /* end of template specialization */
+  (void)db;
+  (void)statement;
+  (void)cnt;
+}
+
+template <typename T, typename... Types>
+static void bindArguments(sqlite3* db, sqlite3_stmt* statement, int cnt, const T& v, const Types&... args) {
+  bindArgument<const T&>(db, statement, cnt, v);
+  bindArguments<const Types&...>(db, statement, cnt + 1, args...);
+}
+
+class SQLiteStatement {
+ public:
+  inline sqlite3_stmt* get() const { return stmt_.get(); }
+  inline int step() const { return sqlite3_step(stmt_.get()); }
+
+  template <typename... Types>
+  static SQLiteStatement prepare(sqlite3* db, const char* zSql, const Types&... args) {
+    sqlite3_stmt* statement;
+
+    if (sqlite3_prepare_v2(db, zSql, -1, &statement, nullptr) != SQLITE_OK) {
+      LOG_ERROR << "Could not prepare statement: " << sqlite3_errmsg(db);
+      throw std::runtime_error("SQLite fatal error");
+    }
+
+    bindArguments<Types...>(db, statement, 1, args...);
+
+    return SQLiteStatement(db, statement);
+  }
+
+ private:
+  SQLiteStatement(sqlite3* db, sqlite3_stmt* stmt) : db_(db), stmt_(stmt, sqlite3_finalize) {}
+
+  sqlite3* db_;
+  std::unique_ptr<sqlite3_stmt, int (*)(sqlite3_stmt*)> stmt_;
+};
+
+// Unique ownership SQLite3 connection
+class SQLite3Guard {
+ public:
+  sqlite3* get() { return handle.get(); }
+  int get_rc() { return rc; }
+
+  SQLite3Guard(const char* path) : handle(nullptr, sqlite3_close), rc(0) {
+    sqlite3* h;
+    rc = sqlite3_open(path, &h);
+    handle.reset(h);
+  }
+
+  int exec(const char* sql, int (*callback)(void*, int, char**, char**), void* cb_arg) {
+    return sqlite3_exec(handle.get(), sql, callback, cb_arg, NULL);
+  }
+
+  std::string errmsg() const { return sqlite3_errmsg(handle.get()); }
+
+  template <typename... Types>
+  SQLiteStatement prepareStatement(const char* zSql, const Types&... args) {
+    return SQLiteStatement::prepare(handle.get(), zSql, args...);
+  }
+
+ private:
+  std::unique_ptr<sqlite3, int (*)(sqlite3*)> handle;
+  int rc;
```

Comment:
add underscore for private memebres

Reply:
Is that actually a strict rule for us? We don't apply it very consistently. Google's style guide has all class member variables add a trailing underscore. We certainly don't do that. I'm okay with any version, but consistency would be nice.

Reply:
We should certainly apply it to new code, if only because 'just following the style guide' is less effort then deciding which bits we like

Reply:
In this case, I kept the case of the original class (which was moved from one file to another).

I refreshed it to follow the right™ way